nuxeo-services/login/nuxeo-platform-login-token-rest/src/test/java/org/nuxeo/ecm/restapi/test/AuthenticationTokensTest.java

```diff
@@ -106,16 +107,79 @@ public void itCanRevokeTokens() throws Exception {
+        }
+
+        // check tokens for current user
+        List<JsonNode> tokens = getTokens();
+        assertEquals(1, tokens.size());
+        JsonNode token = tokens.get(0);
+        assertEquals("app", token.get("application").textValue());
+        assertEquals("device", token.get("deviceId").textValue());
+        assertEquals("rw", token.get("permission").textValue());
+        assertFalse(token.get("creationDate").textValue().isEmpty());
+        assertFalse(token.get("username").textValue().isEmpty());
+    }
+
+    @Test
+    public void itCanCreateJSONTokens() throws Exception {
+        // acquire a token
+        MultivaluedMap<String, String> params = new MultivaluedMapImpl();
+        params.put("application", Collections.singletonList("app"));
+        params.put("deviceId", Collections.singletonList("device"));
+        params.put("permission", Collections.singletonList("rw"));
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "/token", null, params, null, null,
+                new String[] { MediaType.APPLICATION_JSON })) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+            String tokenId = response.getEntity(String.class);
+            assertFalse(tokenId.isEmpty());
+            assertTrue(tokenId.startsWith("\""));
+            assertTrue(tokenId.endsWith("\""));
+            assertTrue(MediaType.APPLICATION_JSON_TYPE.isCompatible(response.getType()));
+        }
+
+        // check tokens for current user
+        List<JsonNode> tokens = getTokens();
+        assertEquals(1, tokens.size());
+        JsonNode token = tokens.get(0);
+        assertEquals("app", token.get("application").textValue());
+        assertEquals("device", token.get("deviceId").textValue());
+        assertEquals("rw", token.get("permission").textValue());
+        assertFalse(token.get("creationDate").textValue().isEmpty());
+        assertFalse(token.get("username").textValue().isEmpty());
+    }
+
+    @Test
+    public void itCanCreateDetailedJSONTokens() throws Exception {
+        // acquire a token
+        MultivaluedMap<String, String> params = new MultivaluedMapImpl();
+        params.put("application", Collections.singletonList("app"));
+        params.put("deviceId", Collections.singletonList("device"));
+        params.put("permission", Collections.singletonList("rw"));
+        params.put("detail", Collections.singletonList("true"));
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "/token", null, params, null, null,
+                new String[] { MediaType.APPLICATION_JSON })) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+            JsonNode respToken = mapper.readTree(response.getEntityInputStream());
+            assertTrue(MediaType.APPLICATION_JSON_TYPE.isCompatible(response.getType()));
+            assertEquals("app", respToken.get("application").textValue());
```

Comment:
Shouldn't we check `respToken.get("token")` too? It's the main info after all

Reply:
Done

Reply:
I don't see a change. Did you push on the correct repo?

Reply:
Yes.  The JSON serializes the token value as "id" and not "token".

Reply:
Ah ok yes `AuthenticationTokenWriter` does that work and uses a different name.